tests/unit/stage/test_imports.py

```diff
@@ -0,0 +1,58 @@
+from dvc.path_info import PathInfo
+from dvc.stage.imports import get_dir_changes, update_import_dir
+
+
+def test_get_dir_changes(tmp_dir, dvc):
+
+    imp = tmp_dir.gen({"datadir": {"foo": "foo", "bar": "bar"}})[0]
+
+    out = tmp_dir / "out"
+
+    stage = dvc.imp_url(str(imp), str(out))
+
+    (tmp_dir / "datadir" / "foo").write_text("test")
+    (tmp_dir / "datadir" / "foo1").write_text("test_1")
+
+    files_to_down, files_to_rem = get_dir_changes(stage)
+    assert tmp_dir / "datadir" / "foo" in files_to_down
+    assert tmp_dir / "datadir" / "foo1" in files_to_down
+    assert tmp_dir / "out" / "foo" in files_to_rem
+
+
+def test_get_dir_changes_nested(tmp_dir, dvc):
+
+    imp = tmp_dir.gen(
+        {"data": {"file": "file", "folder": {"file1": "file1"}}}
+    )[0]
+
+    out = tmp_dir / "out"
+
+    stage = dvc.imp_url(str(imp), str(out))
+
+    (tmp_dir / "data" / "folder" / "file1").write_text("test")
+
+    files_to_down, files_to_rem = get_dir_changes(stage)
+
+    assert tmp_dir / "data" / "folder" / "file1" in files_to_down
+    assert tmp_dir / "out" / "folder" / "file1" in files_to_rem
+
+
+def test_update_import_dir(tmp_dir, dvc, mocker):
+    imp = tmp_dir.gen({"data": {"file3": "foo", "file4": "bar"}})[0]
+
+    out = tmp_dir / "out"
+
+    stage = dvc.imp_url(str(imp), str(out))
+    save_deps = mocker.patch.object(stage, "save_deps")
+    remove = mocker.patch.object(stage.outs[0].fs, "remove")
+    download = mocker.patch.object(stage.deps[0].fs, "download")
+    with mocker.patch(
```

Comment:
A warning here, according to https://pypi.org/project/pytest-mock/

> The purpose of this plugin is to make the use of context managers and function decorators for mocking unnecessary, so it will emit a warning when used as such.